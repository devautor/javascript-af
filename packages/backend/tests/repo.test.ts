import { graphqlTestCall } from './utils/gqlTestClient';
import { prisma } from './utils/createPrismaMock';
import { aggregateMock, ownerMock } from './mocks';

describe('Repo Resolvers', () => {
  prisma.repoesConnection = jest.fn(() => ({
    pageInfo: {
      hasNextPage: false,
      hasPreviousPage: false,
      startCursor: 'cjq0qx0v800050981rha92aq2',
      endCursor: 'cjq0qx0v800050981rha92aq2',
    },
    edges: [
      {
        node: {
          id: 'cjq0qx0v800050981rha92aq2',
          githubName: 'IhaveWings',
          githubOwner: 'LuciferM',
        },
        cursor: 'cjq0qx0v800050981rha92aq2',
      },
      {
        node: {
          id: 'cjq0qx0v800050981rha92qsa',
          githubName: 'WingsAreGone',
          githubOwner: 'Amenadiel',
        },
        cursor: 'cjq0qx0v800050981rha92qsa',
      },
    ],
    aggregate: aggregateMock,
  }));

  prisma.repo = jest.fn(() => ({
    id: 'cjq0qx0v800050981rha92aq2',
    githubName: 'IhaveWings',
    githubOwner: 'LuciferM',
    slug: 'LuciferM/IhaveWings',
    isFeatured: true,
    createdAt: '2018-12-23T10:24:29.259Z',
    updatedAt: '2018-12-23T10:24:29.259Z',
    owner: ownerMock,
  }));

  test('repoConnection pagination', async () => {
    const response = await graphqlTestCall(/* GraphQL */ `
    {
      repoConnection(orderBy:createdAt_DESC) {
        pageInfo {
          hasNextPage
          hasPreviousPage
          startCursor
          endCursor
        }
        edges {
          node {
            id
            githubName
            githubOwner
            owner{
              id
              username
            }
          }
          cursor
        }
        aggregate {
          count
        }
      }
    }
  `);
    expect(prisma.repoesConnection).toHaveBeenCalledWith({
      orderBy: 'createdAt_DESC',
    });
    expect(aggregateMock).toHaveBeenCalled();
    expect(prisma.repo).toHaveBeenCalledWith({
      id: 'cjq0qx0v800050981rha92aq2',
    });
    expect(ownerMock).toHaveBeenCalled();
    expect(response).toMatchSnapshot();
  });

  test('repoBySlug', async () => {
    const response = await graphqlTestCall(/* GraphQL */ `
      {
  repoBySlug(slug: "LuciferM/IhaveWings") {
    id
    slug
    githubName
    githubOwner
    isFeatured
    createdAt
    updatedAt
    owner {
      id
      name
      username
      email
      githubToken
      profilePic
      createdAt
      updatedAt
    }
  }
}
  `);
    expect(prisma.repo).toHaveBeenCalledWith({
      slug: 'LuciferM/IhaveWings',
    });
    expect(ownerMock).toHaveBeenCalled();
    expect(response).toMatchSnapshot();
  });
});
