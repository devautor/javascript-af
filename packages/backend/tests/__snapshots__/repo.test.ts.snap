// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo Resolvers repoBySlug 1`] = `
Object {
  "data": Object {
    "repoBySlug": Object {
      "createdAt": "2018-12-23T10:24:29.259Z",
      "githubName": "IhaveWings",
      "githubOwner": "LuciferM",
      "id": "cjq0qx0v800050981rha92aq2",
      "isFeatured": true,
      "owner": Object {
        "createdAt": "2018-12-21T07:25:09.804Z",
        "email": "iam@thedevil.com",
        "githubToken": "s",
        "id": "cjpxpmpgr000c097495f1kz8a",
        "name": "Lucifer Morningstar",
        "profilePic": "ss",
        "updatedAt": "2018-12-21T07:25:09.804Z",
        "username": "LuciferMorningstar",
      },
      "slug": "LuciferM/IhaveWings",
      "updatedAt": "2018-12-23T10:24:29.259Z",
    },
  },
}
`;

exports[`Repo Resolvers repoConnection pagination 1`] = `
Object {
  "data": Object {
    "repoConnection": Object {
      "aggregate": Object {
        "count": 2,
      },
      "edges": Array [
        Object {
          "cursor": "cjq0qx0v800050981rha92aq2",
          "node": Object {
            "githubName": "IhaveWings",
            "githubOwner": "LuciferM",
            "id": "cjq0qx0v800050981rha92aq2",
            "owner": Object {
              "id": "cjpxpmpgr000c097495f1kz8a",
              "name": "Lucifer Morningstar",
            },
          },
        },
        Object {
          "cursor": "cjq0qx0v800050981rha92qsa",
          "node": Object {
            "githubName": "WingsAreGone",
            "githubOwner": "Amenadiel",
            "id": "cjq0qx0v800050981rha92qsa",
            "owner": Object {
              "id": "cjpxpmpgr000c097495f1kz8a",
              "name": "Lucifer Morningstar",
            },
          },
        },
      ],
      "pageInfo": Object {
        "endCursor": "cjq0qx0v800050981rha92qsa",
        "hasNextPage": false,
        "hasPreviousPage": false,
        "startCursor": "cjq0qx0v800050981rha92aq2",
      },
    },
  },
}
`;
